test/e2e/onCLS-test.js

```diff
@@ -460,23 +463,25 @@ describe('onCLS()', async function () {
     assert.match(cls1.navigationType, /navigate|reload/);
 
     await clearBeacons();
-    await triggerLayoutShift();
-
     await stubForwardBack();
-    await beaconCountIs(1);
```

Comment:
@philipwalton I don't understand this logic and I think it was causing the tests to be flakey.

Why trigger a layout shift before the back/forward? Doesn't that mean it should really be for the previous page (before the "forward")? In which case we'll have another CLS reported for the original page (depending if bigger or not), then a 0-CLS reported for the "forward"? Yet you're only checking 1 CLS below.

IT seems to work a lot of the time locally, but not on CI. I suspect the "depending if bigger or not" is causing issues.

I've simplified this to:
- Load page with layout shifts. Check they are reported.
- Do a forward/back. Check 0 CLS is reported for the "forward" page navigation.
- Do a layout shift and then a visibility change. Check non-0 CLS with 1 layout shift is report for the "back" navigation.

Reply:
> @philipwalton I don't understand this logic and I think it was causing the tests to be flakey.
>
> Why trigger a layout shift before the back/forward? Doesn't that mean it should really be for the previous page (before the "forward")?

Yes, it is for the previous visit. In this test there are 3 visits: 1 for the original navigation and then 2 after a bfcache restore. That's why there are 3 `cls` variables all of which have different IDs (and the test asserts that they are all different).

The reason I wanted to have 3 visits (specifically 2 bfcache restores) is to test that the library doesn't "stop listening" after the first bfcache restore.

AFAICT, all your changes do is remove the layout shifts from the 2nd visit (which is fine; we do have a test to ensure that visits with no CLS are reported as 0, but more tests don't hurt). It's not clear to me why that would reduce flakiness though, since in both cases the logic to trigger a shift and then a navigation is the same. Roughly:

```js
await clearBeacons();
await triggerLayoutShift();

await stubVisibilityChange('hidden'); // or stubBackForward(), which is also a visibility change.
await beaconCountIs(1);
```

Reply:
OK issue was the layoutshifts weren't always finshing before the stubBackForward ran so sometimes they'd bleed into the next page and not report in the previous pages's CLS.

Added a `await nextFrame();` to `triggerLayoutShift` to avoid this happening.